staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation_test.go

```diff
@@ -0,0 +1,352 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/config"
+)
+
+func TestStructure(t *testing.T) {
+	firstResourcePath := resource.Index(0)
+	testCases := []struct {
+		desc string
+		in   *config.EncryptionConfiguration
+		want field.ErrorList
+	}{
+		{
+			desc: "nil encryption config",
+			in:   nil,
+			want: field.ErrorList{
+				field.Required(root, "EncryptionConfiguration can't be nil"),
```

Comment:
pull this into a const, same as with other errors

Reply:
Done.